### ISSUE TITLE
fontenc T2A not found

### ISSUE THREAD

**Issue by l4l** [tectonic-typesetting/tectonic#978]
Basically adding a `\usepackage[T1,T2A]{fontenc}` line (in particular T2A option) breaks the build, i.e steps for reproduction:

```bash
tectonic -X new test
sed -i 's|\\documentclass{article}|\\documentclass{article}\n\\usepackage[T1,T2A]{fontenc}|' src/_preamble.tex
tectonic -X build 2>&1 | rg -v 'warning: accessing absolute path'
```

The output:

```
note: "version 2" Tectonic command-line interface activated
note: Running TeX ...
error: fontenc.sty:112: Font T2A/cmr/m/n/10=larm1000 at 10.0pt not loadable: Metric (TFM) file or installed font not found
error: halted on potentially-recoverable error as specified
```

I'm not really familiar with the tectonic perhaps I need to set special settings? `pdflatex` for the same source works fine for me:

```bash
cat src/_preamble.tex src/index.tex src/_postamble.tex >a.tex
pdflatex a.tex
```

**Comment by ktzanev** [comment]
You don't have to use `fontenc` with `xelatec` and `lualatex`. You have to use `fontspec` (check [this documentation](https://latexref.xyz/fontenc-package.html#:~:text=If%20you%20use%20the%20xelatex%20or%20lualatex%20engine%20then%20instead%20use%20the%20fontspec%20package.) or [this one](https://fr.overleaf.com/learn/latex/International_language_support)). Tectonic is based on `xelatex` so you have to replace `fontenc`  with `fontspec`.

In conclusion, this is not a tectonic specific problem.

If you want to have a code that is compilable with pdflatex and tectonic (xelatex) you can use the following code : 
```latex
\usepackage{iftex}
\ifPDFTeX % PDFLaTeX
  \usepackage[utf8]{inputenc}
  \usepackage[T1,T2A]{fontenc}
  \usepackage{lmodern}
\else % LuaLaTeX & XeLaTeX (tectonic)
  \usepackage{fontspec}
\fi
```

**Comment by mnrvwl** [comment]
Hey @l4l thanks for the great issue 🚀
Thx @ktzanev for the insight and workaround!

While @ktzanev is correct about `XeLaTeX` and `LuaLaTeX` not needing to use `fontenc`, I don't think it's precise to say this is not a `tectonic` specific and the tests I ran make me believe this is still a legitimate issue.

Let's take a simple MWE `test.tex`:

```latex
\documentclass{article}
  \usepackage[T2A]{fontenc}

\begin{document}
  This is a simple example typesetting the \symbol{"C1} symbol.
\end{document}
```

When running `tectonic test.tex`, we reproduce @l4l's error:
```
error: fontenc.sty:112: Font T2A/cmr/m/n/10=larm1000 at 10.0pt not loadable: Metric (TFM) file or installed font not found
error: halted on potentially-recoverable error as specified
```

What if we use XeLaTeX? Let's give it a look:

```console
$ latexmk -pdfxe test.tex
[...]
4142 bytes written
Latexmk: All targets (test.pdf) are up-to-date
```

Same story with LuaLaTeX:
```console
$ latexmk -pdflua test.tex
Latexmk: Log file says output to 'test.pdf'
Latexmk: All targets () are up-to-date
```

### Conclusions

[`fontenc` works well with XeTeX/LuaTeX](https://tex.stackexchange.com/a/31712), and it's intended to do so. Using it is _not necessary_ nor required with XeTeX/LuaTeX, but _it is **compatible**_ with both engines. As per https://github.com/tectonic-typesetting/tectonic/issues/158#issuecomment-378281566 and https://github.com/tectonic-typesetting/tectonic/issues/158#issuecomment-378283864, @pkgw wants to have `tectonic` stay as close as possible with XeTeX, their documents and get at least some pdfTeX documents to still compile correctly. Otherwise, we should document deliberate compatibility break somewhere, as many documents in Cyrillic languages (e.g. Bulgarian, Russian, Serbian, Ukrainian... etc.) intended for pdfTeX are using `T2A`. This should remain open until we decide either way.

Giving workarounds is good! We also have to be mindful that workarounds don't make issues disappear, and that the fact a workaround exists doesn't always let us to wave off an issue and dismiss it.

### Versions
- `Tectonic 0.12.0`
- `TeX Live 2022`
- `LuaHBTeX, 1.15.0`
- `XeTeX 3.141592653-2.6-0.999994`

**Comment by vlasakm** [comment]
See also: https://github.com/tectonic-typesetting/tectonic/issues/17, https://github.com/tectonic-typesetting/tectonic/issues/633, https://github.com/tectonic-typesetting/tectonic/issues/857

@mnrvwl I agree that while using things like `fontenc` should be discouraged in the TeX world in general, there is (in general) no reason why Tectonic shouldn't behave like XeLaTeX does. Though in this case it is a bit harder.

The root cause is that XeLaTeX is instructed to use the font `larm1000`, so it goes looking for it or rather in this case for the metric file `larm1000.tfm`. This file however doesn't normally exist, it is generated by [mktextfm](https://github.com/TeX-Live/texlive-source/blob/trunk/texk/texlive/linked_scripts/texlive/mktextfm) from `larm1000.mf`, which is _also_ generated, though by [mktexmf](https://github.com/TeX-Live/texlive-source/blob/trunk/texk/texlive/linked_scripts/texlive/mktexmf).

Generating the `.mf` file is relatively  easy. Processing it into `.tfm` (metrics) and `.pk` (bitmap) is the hard part since this is done by the Metafont program, which is not included in Tectonic. Maybe it should, because old TeX workflows can depend on it (see for example https://github.com/tectonic-typesetting/tectonic/issues/857).

So yes, this is indeed Tectonic's fault -- it doesn't work like `xelatex` does. Though this `.tfm` and `.mf` generation is transparent to `xelatex` (or any other `tex`) and instead a matter of the distribution (what I show above is TeX Live, MikTeX probably does something different).

Given that  the benefit of XeTeX / Tectonic is the possibility to use Unicode and OpenType (`.otf`, `.ttf`) fonts and introducing the support for Metafont would produce subpar results anyway (bitmap fonts!) I don't think that this will be ever fixed in Tectonic.

**Comment by mnrvwl** [comment]
I have found another example that fails. Its purpose is to automatically convert transliterated text into cyrillic:
```latex
\documentclass{article}
  \usepackage[OT2,T1]{fontenc}
  \usepackage[russian,english]{babel}

\begin{document}
  Write: \foreignlanguage{russian}{Rezoluciya o prolitprosvetrabote}.
\end{document}
```

It compiles without issues under pdfTeX, XeTeX and LuaTeX, but in `tectonic` it will error with:
```console
error: tl.tex:6: Font OT2/cmr/m/n/10=wnr10 at 10.0pt not loadable: Metric (TFM) file or installed font not found
error: halted on potentially-recoverable error as specified
```

@pkgw Instead of creating separate issues for T2A and OT2, I think it would make more sense to have a single one to track both of them. Even though there may be ways to work around the issue in some cases, there are still valid uses where this is not possible. `Tectonic` currently does not support these uses. How would you feel about reopening this?

**Comment by pkgw** [comment]
Hmm, I think it might be better to create a new issue about the possibility of new Tectonic capabilities in this area, in contrast to issues where the answer is basically "don't use `fontenc`". I'll do that now.

**Comment by pkgw** [comment]
OK, I opened #992 on this topic. (Getting close to 1000!)
